kernel/nativecode.ml

```diff
@@ -500,6 +500,7 @@ type mllambda =
   | MLsequence     of mllambda * mllambda
   | MLarray        of mllambda array
   | MLisaccu       of string * inductive * mllambda
+  | MLthunk        of mllambda
```

Comment:
I don't like adding a new primitive, can't we just generate the thunk directly?

Reply:
IDK how to do that without a bunch of extra Obj.magic, and I prefer a new primitive to the Obj.magic.